src/nsolid.h

```diff
@@ -1764,6 +1805,13 @@ void on_configuration_hook_proxy_(std::string info, void* data) {
   (*static_cast<G*>(data))(info);
 }
 
+template <typename G>
+void on_log_write_hook_proxy_(SharedEnvInst inst,
+                              std::string line,
+                              void* data) {
+  (*static_cast<G*>(data))(inst, line);
```

Comment:
Maybe use `std::move() here. I'm not too fond of those 2 string copies per hook call.
```suggestion
  (*static_cast<G*>(data))(inst, std::move(line));
```